From 959bbe4636373bd4e9a18ac470476803cc2087d0 Mon Sep 17 00:00:00 2001
From: Jerry Jelinek <jerry.jelinek@joyent.com>
Date: Thu, 24 Aug 2017 22:46:01 +0000
Subject: [PATCH] OS-6308 fix for OS-6275 blows assert on debug kernel Reviewed
 by: Patrick Mooney <patrick.mooney@joyent.com> Approved by: Patrick Mooney
 <patrick.mooney@joyent.com>

---
 usr/src/uts/common/brand/lx/syscall/lx_pipe.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/usr/src/uts/common/brand/lx/syscall/lx_pipe.c b/usr/src/uts/common/brand/lx/syscall/lx_pipe.c
index cdada58f15..b96727cd8a 100644
--- a/usr/src/uts/common/brand/lx/syscall/lx_pipe.c
+++ b/usr/src/uts/common/brand/lx/syscall/lx_pipe.c
@@ -186,7 +186,9 @@ lx_hd_pipe(intptr_t arg, int flags)
 	 * the pipe capacity is limited to "Fifohiwat" which is a compile-time
 	 * limit set to FIFOHIWAT.
 	 */
+	mutex_enter(&VTOF(vp1)->fn_lock->flk_lock);
 	fifo_fastoff(VTOF(vp1));
+	mutex_exit(&VTOF(vp1)->fn_lock->flk_lock);
 
 	/*
 	 * Set the O_NONBLOCK flag if requested.
-- 
2.21.0

